src/components/sendV2/summary/summary.js

```diff
@@ -24,22 +26,76 @@ class Summary extends React.Component {
       },
     };
 
-    this.prevStep = this.prevStep.bind(this);
-    this.nextStep = this.nextStep.bind(this);
+    this.checkForHardwareWallet = this.checkForHardwareWallet.bind(this);
     this.checkSecondPassphrase = this.checkSecondPassphrase.bind(this);
-    this.amountAndFee = this.amountAndFee.bind(this);
+    this.nextStep = this.nextStep.bind(this);
+    this.prevStep = this.prevStep.bind(this);
+    this.submitTransaction = this.submitTransaction.bind(this);
   }
 
   componentDidMount() {
     const { account } = this.props;
+    let newState = {};
 
     // istanbul ignore else
     if (account && account.secondPublicKey) {
-      this.setState({
+      newState = {
+        ...this.state,
         secondPassphrase: {
           ...this.state.secondPassphrase,
           hasSecondPassphrase: true,
         },
+      };
+    }
+
+    // istanbul ignore else
+    if (account.hwInfo && account.hwInfo.deviceId) {
+      newState = {
+        ...this.state,
```

Comment:
Maybe you don't need this one here? Since the setState will just receive the 2 following keys, it should alter the previous state. Same on line `43`

Reply:
done